feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts

```diff
@@ -0,0 +1,185 @@
+  RoutingService,
+  WindowRef,
+} from '@spartacus/core';
+import { combineLatest, Observable, of, Subscription } from 'rxjs';
+import { filter, map, switchMap } from 'rxjs/operators';
+import { AddedToCartToastConfig } from '../../added-to-cart-toast-config';
+import { CartToastItem, CART_TOAST_STATE } from '../../core/models';
+import { AddedToCartToastService } from '../../core/services';
+
+@Component({
+  selector: 'cx-added-to-cart-toast',
+  templateUrl: './added-to-cart-toast.component.html',
+})
+export class AddedToCartToastComponent implements OnInit, OnDestroy {
+  protected subscription = new Subscription();
+
+  quantity: number;
+
+  customClass?: string;
+
+  @HostBinding('className') baseClass: string;
+
+  headerElement: HTMLElement | null;
+
+  toastContainerClass: string;
+
+  timeout: number;
+
+  toastContainerBaseClass = 'toast-container';
+
+  scrollEventUnlistener: () => void;
+
+  cartAddEntrySuccess$: Observable<CartAddEntrySuccessEvent> = this.eventService.get(
+    CartAddEntrySuccessEvent
+  );
+
+  cartToasts$: Observable<
+    CartToastItem[]
+  > = this.addedToCartToastService.getToasts();
+
+  constructor(
+    protected winRef: WindowRef,
+    protected addedToCartToastConfig: AddedToCartToastConfig,
+    protected addedToCartToastService: AddedToCartToastService,
+    protected eventService: EventService,
+    protected productService: ProductService,
+    protected renderer: Renderer2,
+    protected routerService: RoutingService
+  ) {}
+
+  ngOnInit() {
+    this.headerElement = this.winRef.document.querySelector('header');
+
+    if (!this.customClass) this.customClass = 'cx-added-cart-toast';
+
+    this.baseClass = `${this.customClass}`;
+    this.toastContainerClass = this.toastContainerBaseClass;
+
+    if (this.addedToCartToastConfig.addedToCartToast?.timeout)
+      this.timeout = this.addedToCartToastConfig.addedToCartToast?.timeout;
+    else this.timeout = 3000;
+
+    this.subscription.add(
+      this.cartAddEntrySuccess$
+        .pipe(
+          switchMap((cartEntry) => {
+            return combineLatest([
+              of(cartEntry),
+              this.productService.get(cartEntry.productCode),
+            ]).pipe(
+              filter((result) => result[1] !== undefined),
+              map(([cartEntry, product]) => ({
+                quantityAdded: cartEntry.quantityAdded,
+                product,
+              }))
+            );
+          })
+        )
+        .subscribe(({ quantityAdded, product }) => {
+          let toastContainerClass = `${this.toastContainerBaseClass} `;
+          const toastItem = this.addedToCartToastService.addToast(
+            quantityAdded,
+            product,
+            toastContainerClass
+          );
+          // trigger enter animations with class change
+          setTimeout(() => {
+            toastItem.baseClass = this.getToastStyles();
+          });
+
+          this.scrollEventUnlistener && this.scrollEventUnlistener();
```

Comment:
I spoke with the architect and ask if there isany approach for short if condition in spartacus code and he said that there is no restriction to make it easier for future change mostly everybody I using standard if () {} else {} approach. May you please change it in your code?